sources/common/test/unitTest/EventSignal.cpp

```diff
@@ -38,9 +38,11 @@ static inline std::string getEventSignalPrint()
 /// Create EventSignal.
 static inline SIGNAL::EventSignal getEventSignal()
 {
-    return {DummySignal::DUMMY_NAME,
-            DummySignal::DUMMY_TIMESTAMP,
-            DummySignal::DUMMY_HANDLE};
+    static std::vector<std::unique_ptr<const SIGNAL::SignalDescriptor>> vDesc = {};
+    vDesc.push_back(std::make_unique<const SIGNAL::SignalDescriptor>(DummySignal::DUMMY_NAME, DummySignal::DUMMY_SIZE, DummySignal::DUMMY_TYPE, DummySignal::DUMMY_HANDLE));
+
+    return { vDesc.back().get(),
```

Comment:
I don't get it. Why do you need this vector? Also, unique_ptr is not stoted anywhere so its destructor will delete object.

Reply:
Have anything changed here?

Reply:
As we talked some times ago. It seems to work and I couldn't think of a better solution. If you could come up with any proposal that would be fantastic. As far as I can remember this vector is needed because now we store so called 'SignalDescriptors' which might be common for several Signal objects. They just store common properties.

Reply:
Just fixed that. It wasn't necessary here.